### PR TITLE
Add includes for boost::in_place & remove obsolete test file

### PR DESCRIPTION
**Pull request by seelabs**
Changes needed to let ripple build with boost 1.59

### CODE REVIEW

File: src/ripple/overlay/impl/OverlayImpl.cpp [hunk 1]
```diff
@@ -144,7 +144,6 @@ OverlayImpl::OverlayImpl (
     BasicConfig const& config)
     : Overlay (parent)
     , io_service_ (io_service)
-    , work_ (boost::in_place(std::ref(io_service_)))
```
Comment (vinniefalco) [inline src/ripple/overlay/impl/OverlayImpl.cpp]: It was nice to set up `work_` in the ctor-initializer, is there a way to do it? I won't hold up the PR for it though.

File: src/ripple/overlay/tests/short_read.test.cpp [hunk 2]
```diff
@@ -533,14 +533,14 @@ class short_read_test : public beast::unit_test::suite
...
-        : work_(boost::in_place(std::ref(io_service_)))
-        , thread_(std::thread([this]()
+        : thread_(std::thread([this]()
             {
                 beast::Thread::setCurrentThreadName("io_service");
                 this->io_service_.run();
             }))
         , context_(make_SSLContext())
     {
+        work_.emplace (io_service_);
```
Comment (vinniefalco) [inline src/ripple/overlay/tests/short_read.test.cpp]: LOL.. no. Did tests pass with this change? `io_service_.run()` will return immediately if `work_` is not constructed first. See: http://www.boost.org/doc/libs/1_58_0/doc/html/boost_asio/reference/io_service__work.html

File: src/ripple/peerfinder/sim/Tests.cpp [hunk 3]
```diff
@@ -188,10 +188,10 @@ class Node
         , m_sink (prefix(), journal.sink())
         , m_journal (Journal (m_sink, journal.severity()), Reporting::node)
         , m_next_port (m_config.listening_endpoint.port() + 1)
-        , m_logic (boost::in_place (
-            std::ref (clock), std::ref (*this), std::ref (*this), std::ref (*this), m_journal))
         , m_when_expire (m_network.now() + std::chrono::seconds (1))
     {
+        m_logic.emplace (
+            clock, *this, *this, *this, m_journal);
```
Comment (vinniefalco) [inline src/ripple/peerfinder/sim/Tests.cpp]: I think we should delete this entire file

File: src/ripple/server/tests/Server.test.cpp [hunk 4]
```diff
@@ -48,9 +48,9 @@ class Server_test : public beast::unit_test::suite
 
     public:
         TestThread()
-            : work_(boost::in_place(std::ref(io_service_)))
-            , thread_([&]() { this->io_service_.run(); })
+            : thread_([&]() { this->io_service_.run(); })
         {
+            work_.emplace (io_service_);
```
Comment (vinniefalco) [inline src/ripple/server/tests/Server.test.cpp]: This ain't gonna work, the call to `work_.emplace` must happen before the call to `io_service_.run`.

**Comment by vinniefalco** [comment]
Reviewers: @nbougalis @vinniefalco

**Comment by seelabs** [comment]
@vinniefalco @nbougalis Hold off on reviewing this any more. I'll push a fix tomorrow morning (`in_place_factory` is still in 1.59, it just wasn't finding it and I read a stack overflow answer too quickly).

FYI: I did run the current code through the usual unit/npm/sync/stop tests.

**Comment by vinniefalco** [comment]
@seelabs I'd like to understand how two of those tests could succeed even when the thread is not processing the `io_service` queue.

**Comment by seelabs** [comment]
@vinniefalco I looked into why unittest succeeded. I ran the tests like this: `rippled --unittest`. All tests succeed. However, the problematic tests are marked `BEAST_DEFINE_TESTSUITE_MANUAL` and are not run in that scenario. If I run `rippled --unittest=short_read` the test hangs forever, as you predicted.

**Comment by vinniefalco** [comment]
derp

**Comment by seelabs** [comment]
@vinniefalco @nbougalis Pushed changes, open again for review. Note the `Server` unittest fails, but it fails on develop as well and is unrelated to these changes (The `Server` test is not run by default).

**Comment by vinniefalco** [comment]
@seelabs  Should we change the `Server` test to use a higher port number so it doesn't need root?

**Comment by seelabs** [comment]
@vinniefalco I just pushed a change `Server` to use port 1025, which did not require root to run.

**Comment by vinniefalco** [comment]
:+1:

**Pull request closed without merging**